src/NodeApi/Runtime/NodeJSRuntime.cs

```diff
@@ -0,0 +1,2277 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using static Microsoft.JavaScript.NodeApi.JSNativeApi.Interop;
+
+namespace Microsoft.JavaScript.NodeApi.Runtime;
+
+public unsafe class NodejsRuntime : JSRuntime
```

Comment:
Instead of re-writing the whole low-level API from scratch, it would be better to implement it as a layer on top of the existing JSNativeApi.Interop.cs.
Loading of native functions from a specific DLL must be static anyway. 
The new virtual interface on top of it would allow the required flexibility to either use specific DLL API directly or add tracing on top of it. We can also load it from different DLL.

Reply:
> Instead of re-writing the whole low-level API from scratch, it would be better to implement it as a layer on top of the existing JSNativeApi.Interop.cs.

I considered it more of a refactor than a rewrite. I don't see a benefit to having another lower layer.

> Loading of native functions from a specific DLL must be static anyway.

Yes, but this enables different threads to bind functions from different DLLs or different versions of a DLL. So an application could host multiple JS engines using multiple versions of `libnode.dll`. (Why would anyone want to do that? IDK, but I always find that avoiding static data allows for more flexibility.)

Reply:
I wonder what is your approach for the .Net Framework 4.8? One of the reasons we had these static points is to address the delegate lifetime issues.

Reply:
Regarding loading multiple dlls: one scenario as we have it in RN for Windows is to load V8 and Hermes DLL. It is quite a valid point. In C++ I had to use templates that allowed me to have static pointers and still have them to be different per JS engine. I agree with you that this new approach is much simpler and if we do not have much overhead, then it is great.

Reply:
The delegate lifetime issue was when we'd pass an unmanaged delegate to native code without holding onto it in managed code. We dealt with that for callbacks, but it was never an issue with the regular native APIs.

Tests are passing on .NET 4.7.2 with the latest update.